pruntime/enclave/src/contracts/diem.rs

```diff
@@ -0,0 +1,281 @@
+    fn id(&self) -> contracts::ContractId { contracts::DIEM }
+
+    fn handle_command(&mut self, _origin: &chain::AccountId, _txref: &TxRef, _cmd: Command) -> TransactionStatus {
+        TransactionStatus::Ok
+    }
+
+    fn handle_query(&mut self, origin: Option<&chain::AccountId>, req: Request) -> Response {
+        let inner = || -> Result<Response, Error> {
+            match req {
+                Request::AccountData { account_data_b64} => {
+                    println!("account_data_b64: {:?}", account_data_b64);
+                    let account_data = base64::decode(&account_data_b64).unwrap();
+                    let account_info: AccountInfo = bcs::from_bytes(&account_data).unwrap();
+                    println!("account_info:{:?}", account_info);
+                    let exist = self.accounts.iter().any(|x| x.address == account_info.address);
+                    if !exist {
+                        self.accounts.push(account_info);
+                    }
+
+                    Ok(Response::AccountData { size: self.accounts.len() as u32 })
+                }
+
+                Request::VerifyTransaction {account_address, transaction_with_proof_b64} => {
+                    println!("transaction_with_proof_b64: {:?}", transaction_with_proof_b64);
+
+                    let address = AccountAddress::from_hex_literal(&account_address)
+                        .map_err(|_|Error::Other("Bad account address".to_string()))?;
+                    if !self.accounts.iter().any(|x| x.address == address) {
+                        println!("Not a contract's account address");
+                        return Ok(Response::VerifyTransaction { total: 0, verified: false });
+                    }
+
+                    let mut transactions: Vec<Transaction> = self.transactions.get(&account_address).unwrap_or(&Vec::new()).to_vec();
+                    let mut total: u32 = transactions.len() as u32;
+
+                    let proof_data = base64::decode(&transaction_with_proof_b64).unwrap_or(Vec::new());
+                    let transaction_with_proof: TransactionWithProof = bcs::from_bytes(&proof_data)
+                        .map_err(|_| Error::Other("Bad transaction with proof".to_string()))?;
+                    println!("transaction_with_proof:{:?}", transaction_with_proof);
+
+                    let transaction: Transaction = match bcs::from_bytes(&transaction_with_proof.transaction_bytes) {
+                        Ok(tx) => tx,
+                        Err(_) => {
+                            println!("decode transaction error");
+                            return Ok(Response::VerifyTransaction { total, verified: false });
+                        }
+                    };
+
+                    let signed_tx: SignedTransaction = transaction.as_signed_user_txn().unwrap().clone();
+                    if signed_tx.raw_txn.sender != address {
+                        let mut found = false;
```

Comment:
Add comment: "// Incoming tx doesn't need to be sequential"

Reply:
Done